webui/react/src/stores/userSettings.tsx

```diff
@@ -141,8 +145,43 @@ class UserSettingsStore extends PollingStore {
     }
   }
 
+  /**
+   * This updates the value of a setting and persists it for future sessions.
+   * @param type The type of the value or an encoder of the value to JSON.
+   * @param key Unique key to store and retrieve the settings
+   * @param fn Function to update the value of the setting at `key`
+   */
+  public update<T extends t.Props, U = t.TypeOfProps<T>>(
+    type: t.TypeC<T>,
+    key: string,
+    fn: (value: U | undefined) => U,
+  ): void;
+  public update<T>(type: t.Type<T, Json>, key: string, fn: (value: T | undefined) => T): void {
```

Comment:
@ashtonG What is the overload for? Also note that you need to duplicate this last function definition if you want it to be visible to callers.

Reply:
I experimented with removing it and I noticed the test breaks because it uses `undefined` in the type and `undefined` isn't actually valid JSON. `undefined` does _almost_ serialize correctly with `JSON.stringify` and `JSON.parse`

```typescript
JSON.stringify({ a: undefined }) // becomes "{}"
// Therefore
const foo = { a: undefined };
_.isEqual(foo.a, JSON.parse(JSON.stringify(foo)).a) // true
_.isEqual(foo, JSON.parse(JSON.stringify(foo))) // false
```
The equality holds up when you inspect each member individually but fails when you inspect the object as a whole. If we really do want to use `undefined` though we can use `partial`.